zio-schema-json/shared/src/main/scala/zio/schema/codec/JsonCodec.scala

```diff
@@ -104,13 +104,13 @@ object JsonCodec {
   def jsonEncoder[A](cfg: JsonCodec.Config)(schema: Schema[A]): ZJsonEncoder[A] =
     JsonEncoder.schemaEncoder(schema, cfg)
 
-  def jsonDecoder[A](schema: Schema[A]): ZJsonDecoder[A] =
+  def jsonDecoder[A](schema: Schema[A])(implicit config: JsonCodec.Config): ZJsonDecoder[A] =
```

Comment:
Changes are binary incompatible.
Config should not be implicit.
You should not reuse the same config value, since these are two different features. They should be configurable independent of each other.